st2api/st2api/controllers/resource.py

```diff
@@ -246,6 +246,25 @@ def _get_from_model_kwargs_for_request(self, request):
         """
         return self.from_model_kwargs
 
+    def _get_one_by_scope_and_name(self, scope, name, from_model_kwargs=None):
+        """
+        Retrieve an item given scope and name. Only KeyValuePair now has concept of 'scope'.
+
+        :param scope: Scope the key belongs to.
+        :type scope: ``str``
+
+        :param name: Name of the key.
+        :type name: ``str``
+        """
+        instance = self.access.get_by_scope_and_name(scope=scope, name=name)
+        if not instance:
+            return instance
```

Comment:
`_get_one_*` methods usually return 404 (`pecan.abort`) if a object is not found (and `_get_by_*` throw `StackStormDBObjectNotFoundError` if an object is not found, and `_get_one` methods usually call `_get_by` methods).

We should do the same here for consistency and also add corresponding test case.